modules/unsupported/wfs-ng/src/main/java/org/geotools/data/wfs/internal/WFSException.java

```diff
@@ -25,9 +29,54 @@
  */
 public class WFSException extends IOException {
     private static final long serialVersionUID = -2828901359361793862L;
+    
+    public static final class ExceptionDetails implements Serializable {
```

Comment:
Please javadoc public class as this forms API, looking for class name, responsibility and relationships (how it is used by whom).

I am not sure why this needs to be an inner class, is it so the fields of WFSException can be serialized as a distinct object?

Reply:
I will add some javadocs, as you suggested.

I have chosen an inner class because the ExceptionDetails are so closely related to the WFSException, like a composition. So everything is in one place. The ExceptionDetails are not expected to be used outside of this class, by the gt-wfs-ng module.
It has nothing to do with serialization.
But I am fine to make the class top level, if you like.